src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -282,28 +283,24 @@ export class PersistentTerminalProcess extends Disposable {
 			this.shutdown(true);
 		}, LocalReconnectConstants.ReconnectionShortGraceTime));
 
-		// TODO: Bring back bufferer
-		// this._bufferer = new TerminalDataBufferer((id, data) => {
-		// 	const ev: IPtyHostProcessDataEvent = {
-		// 		type: 'data',
-		// 		data: data
-		// 	};
-		// 	this._events.fire(ev);
-		// });
-
 		this._register(this._terminalProcess.onProcessReady(e => {
 			this._pid = e.pid;
 			this._cwd = e.cwd;
 			this._onProcessReady.fire(e);
 		}));
 		this._register(this._terminalProcess.onProcessTitleChanged(e => this._onProcessTitleChanged.fire(e)));
 		this._register(this._terminalProcess.onProcessShellTypeChanged(e => this._onProcessShellTypeChanged.fire(e)));
-		// Buffer data events to reduce the amount of messages going to the renderer
-		// this._register(this._bufferer.startBuffering(this._persistentProcessId, this._terminalProcess.onProcessData));
-		this._register(this._terminalProcess.onProcessData(e => this._recorder.recordData(e)));
-		this._register(this._terminalProcess.onProcessExit(exitCode => {
-			// this._bufferer.stopBuffering(this._persistentProcessId);
-		}));
+
+		// Data buffering to reduce the amount of messages going to the renderer
+		this._bufferer = new TerminalDataBufferer((_, data) => this._onProcessData.fire({ data: data, sync: true }));
+		this._register(this._bufferer.startBuffering(this._persistentProcessId, this._terminalProcess.onProcessData));
+		this._register(this._terminalProcess.onProcessExit(() => this._bufferer.stopBuffering(this._persistentProcessId)));
+
+		// Buffered events should flush when a resize occurs
+		this._register(this._terminalProcess.onResize(() => this._bufferer.flushBuffer(this._persistentProcessId)));
```

Comment:
This call goes through `PersistentTerminalProcess.resize` originally so we could flush there instead of creating a new event